### PR TITLE
[Merged by Bors] - feat: Real.exists_natCast_add_one_le_pow_of_one_le

### PR DESCRIPTION
**Pull request by pechersky**
Necessary for arguments in approaching a power from above

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Real/Archimedean.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Real/Archimedean.lean:341]:
```suggestion
  obtain ⟨k, posk, hk⟩ : ∃ k : ℕ, 0 < k ∧ 1 / k + 1 < a := by
```
(this does not require further changes)

**Review by Ruben-VandeVelde — APPROVED** [review]
lgtm otherwise

**Comment by Ruben-VandeVelde** [comment]
maintainer merge

**Review by riccardobrasca — COMMENTED** [review]
The fact that `exists_natCast_add_one_lt_pow_of_one_lt` is missing really means that we don't know the limit of `(m+1)/a^m`?

**Comment by riccardobrasca** [comment]
Anyway LGTM, thanks!

bors d+

**Comment by pechersky** [comment]
I looked for this limit, or anything like mul_le_pow which would be necessary for it, but couldn't find it. The closest thing I found was Bernoulli's inequality at `one_add_mul_self_lt_rpow_one_add`.

@riccardobrasca would you like me to add the tendsto statement? I can strengthen the exists to say that all n greater than m also satisfy this inequality. I think that statement would need a lemma that would look like a + b < a * b, which I couldn't find either.

**Comment by riccardobrasca** [comment]
I don't have time anymore today, but we can just merge this and add the limit later if we need it. Thanks!

**Pull request closed without merging**
